staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -660,6 +722,46 @@ See https://kubernetes.io/docs/reference/using-api/server-side-apply/#conflicts`
 	return nil
 }
 
+func (o *ApplyOptions) calculateSSAMigrationPatch(obj runtime.Object) ([]byte, error) {
```

Comment:
I would move this whole function into client-go for easier consumption by other actors that want to migrate.